src/NSwag.SwaggerGeneration.AspNetCore.Tests/AspNetCoreToSwaggerGenerationTests.cs

```diff
@@ -0,0 +1,765 @@
+
+            //// Assert
+            var operation = Assert.Single(document.Operations, o => o.Path == "/" + nameof(ControllerWithParameters.FromHeaderParameter)).Operation;
+            var parameter = Assert.Single(operation.Parameters);
+            Assert.Equal(SwaggerParameterKind.Header, parameter.Kind);
+            Assert.Equal("parameter1", parameter.Name);
+        }
+
+        [Fact]
+        public async Task FromBodyParametersAreDiscovered()
+        {
+            //// Arrange
+            var generator = new AspNetCoreToSwaggerGenerator(new AspNetCoreToSwaggerGeneratorSettings());
+            var apiDescriptions = GetApiDescriptionGroups(typeof(ControllerWithParameters));
+
+            //// Act
+            var document = await generator.GenerateAsync(apiDescriptions);
+
+            //// Assert
+            var operation = Assert.Single(document.Operations, o => o.Path == "/" + nameof(ControllerWithParameters.FromBodyParameter)).Operation;
+            var parameter = Assert.Single(operation.Parameters);
+            Assert.Equal(SwaggerParameterKind.Body, parameter.Kind);
+            Assert.Equal("model", parameter.Name);
+            Assert.True(parameter.IsRequired);
+        }
+
+        [Fact]
+        public async Task FormFileParametersAreDiscovered()
+        {
+            //// Arrange
+            var generator = new AspNetCoreToSwaggerGenerator(new AspNetCoreToSwaggerGeneratorSettings());
+            var apiDescriptions = GetApiDescriptionGroups(typeof(ControllerWithParameters));
+
+            //// Act
+            var document = await generator.GenerateAsync(apiDescriptions);
+
+            //// Assert
+            var operation = Assert.Single(document.Operations, o => o.Path == "/" + nameof(ControllerWithParameters.FileParameter)).Operation;
+            var parameter = Assert.Single(operation.Parameters);
+            Assert.Equal(SwaggerParameterKind.FormData, parameter.Kind);
+            Assert.Equal("multipart/form-data", Assert.Single(operation.Consumes));
+        }
+
+        [Fact]
+        public async Task ComplexQueryParametersAreProcessed()
+        {
+            //// Arrange
+            var generator = new AspNetCoreToSwaggerGenerator(new AspNetCoreToSwaggerGeneratorSettings());
+            var apiDescriptions = GetApiDescriptionGroups(typeof(ControllerWithParameters));
+
+            //// Act
+            var document = await generator.GenerateAsync(apiDescriptions);
+
+            //// Assert
+            var operation = Assert.Single(document.Operations, o => o.Path == "/" + nameof(ControllerWithParameters.ComplexFromQueryParameter)).Operation;
+            Assert.Collection(
+                operation.Parameters.OrderBy(p => p.Name),
+                parameter =>
+                {
+                    Assert.Equal(nameof(ComplexModel.Header), parameter.Name);
+                    Assert.Equal(SwaggerParameterKind.Query, parameter.Kind);
```

Comment:
@rynowak in Mvc 1.0, sub properties of complex models always show up with a BindingSource of `ModelBinder` even if the property has specific binding source annotations. I don't see it in 2.0. Any guesses on what's up?